core/scripts.py

```diff
@@ -2861,16 +2846,28 @@ def process_task(self, task: DeferredTask):
             for file in files:
                 os.remove(file.name)
 
-    def generate_report(self, data_source_name: str, library_id: int) -> str:
-        """Generate a csv file and return the file path"""
-        with tempfile.NamedTemporaryFile(
-            "w",
-            delete=False,
-        ) as temp:
+    def prepare_email_attachment(self, attachments, file_path, attachment_name, files):
+        """
+        extract contents of files and prepare in a dictionary of email attachments
+        """
+        with open(file_path) as the_file:
+            attachments[attachment_name] = the_file.read()
+            files.append(the_file)
```

Comment:
After you exit the `with` here, this is no longer a valid file handle, so it seems dangerous to add the closed file handle to the files list.